### PR TITLE
Add client certificate support to OpenSSL

### PR DESCRIPTION
**Pull request by anrossi**
Add client certificate support to OpenSSL, enable client certificate tests to run on OpenSSL, and also allow servers to require clients to send a certificate for authentication.
Normalize errors between Schannel and OpenSSL for when no client certificate is presented, but one is required.

### CODE REVIEW

**Comment by nibanks** [comment]
Can you please add a PR description. Is this this whole PR? Shouldn't you need to be enabling tests too? What about the `QUIC_CREDENTIAL_FLAG_REQUIRE_CLIENT_AUTHENTICATION` flag?

**Comment by nibanks** [comment]
This PR should remove the note in https://github.com/microsoft/msquic/blob/main/docs/Platforms.md#linux about not yet supporting client certs with openssl.

File: src/platform/tls_openssl.c [hunk 1]
```diff
@@ -202,28 +226,49 @@ CxPlatTlsCertificateVerifyCallback(
...
+                    TlsContext->SecConfig->Flags,
+                    IsDeferredValidationOrClientAuth?
+                        (uint32_t*)&ValidationResult :
+                        NULL);
+
+            if (!CertificateVerified) {
+                X509_STORE_CTX_set_error(x509_ctx, X509_V_ERR_CERT_REJECTED);
+            }
+        } else {
+            CertificateVerified = X509_verify_cert(x509_ctx);
```
[the remark this answers is not in the document]
Comment (anrossi) [reply src/platform/tls_openssl.c:260]: In the case where the server asks for client certificate, but the client doesn't send one, I've opened issue #1953. I believe both Schannel, and OpenSSL (in this PR) will fail the connection if a client certificate is required, and none presented.

**Review by nibanks — APPROVED** [review]
LGTM. Please make sure deferred tests are enabled too (and remove config flag for disabling).

**Comment by wfurt** [comment]
seems like disabled client certificate tests are  passing now on Linux for .NET with this change. 
thanks @anrossi

**Pull request merged**